crates/typst-cli/src/download.rs

```diff
@@ -0,0 +1,204 @@
+use std::collections::VecDeque;
+use std::io::{self, ErrorKind, Read, Stdout, Write};
+use std::time::{Duration, Instant};
+
+use ureq::Response;
+
+/// Keep track of this many past download amounts
+const DOWNLOAD_TRACK_COUNT: usize = 5;
+
+/// A wrapper around [`ureq::Response`] that reads the response body in chunks
+/// over a websocket and displays statistics about its progress.
+///
+/// Downloads will _never_ fail due to statistics failing to print, print errors
+/// are silently ignored.
+///
+/// [`ureq::Response`]: https://docs.rs/ureq/2.7.1/ureq/struct.Response.html
```

Comment:
I _think_ this should not be necessary. rustfmt should autolink this.

Reply:
How can I check this? I didn't it could do that.

Reply:
Is it just me or is the cli excluded? I can generate docs for each crate, like `cargo doc -p typst-library`, but for the cli I get: no crates with documentation.

Reply:
> `cargo doc --all --open` and then check whether it works.

It works 👍🏻